From b3e632d087f9a555df9ca973b204da6ee947e5d9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 04 Nov 2020 23:35:29 +0200
Subject: [PATCH] [SW-12754] gaudi: security: condition pci sec dis

The preboot_security_pci_disable() function undos the settings done by
BTL to protect the device. It should be called in case the security is
disabled on the device.
While there was still work in progress on implementing full security
functionality, we kept the preboot_security_pci_disable() call
unconditional, so we would not accidentally brick a device.
Now, that the security settings are developed and overwrite the BTL
settings, we can safely move the preboot_security_pci_disable() disable
call to run in case the device security is disabled.

Change-Id: I7d78c0bbe72d577b9105e228985382574c14ec8c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 3877a76..4007501 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1322,20 +1322,10 @@
 	if (is_sec_en) {
 		preboot_puts("Security is enabled\n");
 	} else {
+		preboot_security_pci_disable();
 		preboot_puts("Security is disabled\n");
-		/* TODO: add preboot_security_pci_disable(); */
 	}
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SECURITY_EN, is_sec_en);
-
-	/*
-	 * TODO: Remove this call once it's added to the security enable check
-	 *
-	 *	 Meanwhile, keep the below call until we have finished
-	 *	 initializing all the security and there is no worry
-	 *	 to brick a device.
-	 */
-	preboot_puts("Disabling security temporarily till all implemented\n");
-	preboot_security_pci_disable();
 
 	preboot_secure_cpu();
 	preboot_secure_psoc();
